convokit/model/corpus.py

```diff
@@ -33,124 +41,227 @@ class Corpus:
         # private storage
-        self._vector_matrices = dict()
+        self._vector_matrices = {}
 
+        speakers_data = defaultdict(dict)
         convos_data = defaultdict(dict)
-        if exclude_utterance_meta is None: exclude_utterance_meta = []
-        if exclude_conversation_meta is None: exclude_conversation_meta = []
-        if exclude_speaker_meta is None: exclude_speaker_meta = []
-        if exclude_overall_meta is None: exclude_overall_meta = []
-
-        # Construct corpus from file or directory
-        if filename is not None:
-            if disable_type_check: self.meta_index.disable_type_check()
-            if os.path.isdir(filename):
-                utterances = load_uttinfo_from_dir(filename, utterance_start_index,
-                                                   utterance_end_index, exclude_utterance_meta)
-
-                speakers_data = load_speakers_data_from_dir(filename, exclude_speaker_meta)
-                convos_data = load_convos_data_from_dir(filename, exclude_conversation_meta)
-                load_corpus_meta_from_dir(filename, self.meta, exclude_overall_meta)
-
-                with open(os.path.join(filename, "index.json"), "r") as f:
-                    idx_dict = json.load(f)
-                    self.meta_index.update_from_dict(idx_dict)
-
-                # load all processed text information, but don't load actual text.
-                # also checks if the index file exists.
-                # try:
-                #     with open(os.path.join(filename, "processed_text.index.json"), "r") as f:
-                #         self.processed_text = {k: {} for k in json.load(f)}
-                # except:
-                #     pass
-
-                # unpack binary data for utterances
-                unpack_binary_data_for_utts(utterances, filename, self.meta_index.utterances_index,
-                                            exclude_utterance_meta, KeyMeta)
-                # unpack binary data for speakers
-                unpack_binary_data(filename, speakers_data, self.meta_index.speakers_index, "speaker",
-                                   exclude_speaker_meta)
-
-                # unpack binary data for conversations
-                unpack_binary_data(filename, convos_data, self.meta_index.conversations_index, "convo",
-                                   exclude_conversation_meta)
-
-                # unpack binary data for overall corpus
-                unpack_binary_data(filename, self.meta, self.meta_index.overall_index, "overall", exclude_overall_meta)
-
-            else:
-                speakers_data = defaultdict(dict)
-                convos_data = defaultdict(dict)
-                utterances = load_from_utterance_file(filename, utterance_start_index, utterance_end_index)
 
-            self.utterances = dict()
-            self.speakers = dict()
+        if from_corpus is not None:
+            assert utterances is None
```

Comment:
do we need this assert? Agreed that utterances *should* be None, but there's no harm in simply ignoring it, right?